tests/kafkatest/tests/streams/streams_broker_bounce_test.py

```diff
@@ -91,6 +91,7 @@ class StreamsBrokerBounceTest(Test):
     def __init__(self, test_context):
         super(StreamsBrokerBounceTest, self).__init__(test_context)
         self.replication = 3
+        self.num_kafka_nodes = self.replication
```

Comment:
Setting `num_kafka_nodes` to `replication` does not sound correct to me. The replication factor would be smaller than the number of node.

Reply:
`num_kafka_nodes` is only actually used for the underlying check here, we are indeed already setting up the cluster as

```
self.kafka = KafkaService(self.test_context, num_nodes=self.replication, zk=self.zk, topics=self.topics)
```

below, and here it is more like "remembering the num_nodes here". As I suggested, instead of checking the sum of topics (and hence needing this parameter) we could be more strict and check that for each broker node, the listed topics include all the required topic names.